src/rules.jl

```diff
@@ -78,6 +78,33 @@ function (::typeof(frule_kwfunc))(kws::Any, ::typeof(frule), ::RuleConfig, args.
     return frule_kwfunc(kws, frule, args...)
 end
 
+struct ProductTangent{P}
+    partials::P
+end
```

Comment:
How much is gained from this kind of chunking? I mean storing several partials in some tuple. For reverse mode, nothing, I think. For forward mode, maybe `derivatives_using_output` has the same benefits.

The big gain seems to be from making a solid matrix to represent many vectors, and e.g. getting matrix multiplication. So I think the initial design ought to make that work.

Reply:
This design does this (I think). `partials` here can either be a `Tuple{Tuple}` or an `eachrow(Matrix)`

Reply:
Ok, I guess a vector of vectors can have the same meaning as an `EachCol`. Just a different path, by dispatch on `ProductTangent{<:EachSlice}`.